file-shuffler/bash.sh

```
@@ -61,128 +61,136 @@ for category in "${categories[@]}"; do
-            
-            if [[ -n "$rand_time" ]] && touch -t "$rand_time" "$file" 2>/dev/null; then
-                ((timestamp_success_count++))
-            else
-                echo "WARNING: Failed to modify timestamp for \"$(basename "$file")\""
-            fi
+    temp_file_list_final=$(mktemp)
+    find "$category_path" -maxdepth 1 -type f -name "$category-*.*" -print > "$temp_file_list_final"
+
+    while IFS= read -r file; do
+        if [ "$OS" = "Darwin" ]; then
+            rand_time=$(date -v-"$((RANDOM % 10))"d -v"$((RANDOM % 24))"H -v"$((RANDOM % 60))"M +"%Y%m%d%H%M" 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] In POSIX sh, RANDOM is undefined.

(SC3028)